CONTRIBUTING.md

```diff
@@ -129,9 +150,43 @@ Now cd to ~/repos/NVIDIA/spark-rapids and build per
 $ ./build/buildall
 ```
 
-Prior to switching to another spark-rapids-jni or submiting a PR to NVIDIA/spark-rapids-jni,
-we should undo the cudf submodule modification.
+Since we rely on local Maven cache we need to pay extra attention to make sure that
+the final rapids-4-spark artifact included locally built dependencies as opposed to
+CI-built snapshot dependencies from the remote Maven repo. This may happen if Maven
+is invoked with `--offline` or `--no-snapshot-updates` option due to IDE-Maven
```

Comment:
The wording of this implies it is only a possibility if specifying `--offline` or `--no-snapshot-updates`.  Maybe "This may happen even if"?